src/Knp/DictionaryBundle/DependencyInjection/Compiler/DictionaryBuildingPass.php

```diff
@@ -16,44 +16,37 @@ class DictionaryBuildingPass implements CompilerPassInterface
      */
     public function process(ContainerBuilder $container)
     {
-        $config       = $container->getParameter('knp_dictionary.configuration');
-        $class        = $container->getParameter('knp_dictionary.dictionary.dictionary.class');
-        $dictionaries = $config['dictionaries'];
-        $registry     = $container
-            ->getDefinition('knp_dictionary.dictionary.dictionary_registry')
-        ;
+        $config = $container->getParameter('knp_dictionary.configuration');
 
-        foreach ($dictionaries as $name => $dictionary) {
-            $definition = $this->createDefinition($class, $name, $dictionary);
-            $registry->addMethodCall('set', array($name, $definition));
+        foreach ($config['dictionaries'] as $name => $dictionary) {
             $container->setDefinition(
                 sprintf('knp_dictionary.dictionary.%s', $name),
-                $definition
+                $this->createDefinition($name, $dictionary)
             );
         }
     }
 
     /**
-     * @param string $class
-     * @param string $class
+     * @param string $name
      * @param array  $dictionary
      *
      * @return Definition
      */
-    private function createDefinition($class, $name, array $dictionary)
+    private function createDefinition($name, array $dictionary)
```

Comment:
Be careful, the Knp\DictionaryBundle\Dictionary\Dictionary use statement is out of date now